### PR TITLE
Fix/27634: Improve private notes flow

### PR DESCRIPTION
**Pull request by DylanDylann**
### Details

- Improvements to private notes
### Fixed Issues

$ https://github.com/Expensify/App/issues/27634
PROPOSAL: https://github.com/Expensify/App/issues/27634#issuecomment-1778634010

### Tests

a. Case 1: User did not have not in report yet
1. Open expensify app
2. Open any report
3. Open report detail screen
4. Click on "Private notes" 
5. User is navigated to edit note screen
6. Click on save, user is navigate to report detail screen

b. Case 2: User already had a note  in report
1. Open expensify app
2. Open any report
3. Open report detail screen
4. Click on "Private notes" 
5. User is navigated to private note list screen
6. Click on the current user `s private note, user is navigated to edit note screen
- [x] Verify that no errors appear in the JS console

### QA Steps

a. Case 1: User did not have not in report yet
1. Open expensify app
2. Open any report
3. Open report detail screen
4. Click on "Private notes" 
5. User is navigated to edit note screen
6. Click on save, user is navigate to report detail screen

b. Case 2: User already had a note  in report
1. Open expensify app
2. Open any report
3. Open report detail screen
4. Click on "Private notes" 
5. User is navigated to private note list screen
6. Click on the current user `s private note, user is navigated to edit note screen
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[Screencast from 15-11-2023 18:11:41.webm](https://github.com/Expensify/App/assets/141406735/02d6f5f9-9dfa-43f8-94d7-b4f1485665f4)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[Screencast from 15-11-2023 16:57:26.webm](https://github.com/Expensify/App/assets/141406735/753effb2-2871-4e47-b4a5-d4fbb9c9a16f)

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

[Screencast from 15-11-2023 18:03:45.webm](https://github.com/Expensify/App/assets/141406735/9d66788d-10a1-4aeb-8aa4-86e5c7565745)

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

[Screencast from 15-11-2023 15:28:20.webm](https://github.com/Expensify/App/assets/141406735/552132eb-d653-461e-91d4-ce6763149fd0)

</details>

<details>
<summary>MacOS: Desktop</summary>

[Screencast from 15-11-2023 17:23:43.webm](https://github.com/Expensify/App/assets/141406735/abad7cbe-f17b-463d-a43a-b0fe6cb67a3a)

</details>

### CODE REVIEW

File: src/pages/PrivateNotes/PrivateNotesListPage.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
+import {some} from 'lodash';
```
Comment (ArekChr) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:1]: Let's use underscore instead

File: src/pages/PrivateNotes/PrivateNotesListPage.js [hunk 2]
```diff
@@ -97,14 +99,20 @@ function PrivateNotesListPage({report, personalDetailsList, session}) {
...
+                action: () => Navigation.navigate(ROUTES.PRIVATE_NOTES_EDIT.getRoute(report.reportID, accountID)),
                 brickRoadIndicator: privateNoteBrickRoadIndicator(accountID),
+                note: lodashGet(privateNote, 'note', ''),
+                disabled: Number(session.accountID) !== Number(accountID),
             }))
             .value();
     }, [report, personalDetailsList, session, translate]);
 
+    useEffect(() => {
+        if (some(privateNotes, (item) => item.note)) {
```
Comment (ArekChr) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:111]:
```suggestion
        if (_.some(privateNotes, (item) => item.note)) {
```

File: src/pages/PrivateNotes/PrivateNotesListPage.js (hunk 1 above)
Comment (DylanDylann) [reply src/pages/PrivateNotes/PrivateNotesListPage.js]: Updated

File: src/pages/PrivateNotes/PrivateNotesListPage.js (hunk 2 above)
Comment (DylanDylann) [reply src/pages/PrivateNotes/PrivateNotesListPage.js]: updated

**Comment by ArekChr** [comment]
@DylanDylann Please update the branch with the latest main testing...

**Comment by ArekChr** [comment]
@DylanDylann, I have noticed an issue. When we try to delete some private notes by setting them to empty and then save, it unexpectedly redirects back to edit mode. Could you please look into this?

cc: @techievivek, what expected behaviour should be in this case? 


https://github.com/Expensify/App/assets/24796318/1311ae9b-94c5-4fcf-958f-ff6cc925c59b

**Comment by techievivek** [comment]
We would take them to the details page or profile page, whichever page they came from? CC @trjExpensify for cross-checking.

**Comment by trjExpensify** [comment]
Yeah, interesting. I say return to the profile page if there's no notes to preview upon deleting one.

**Comment by DylanDylann** [comment]
@ArekChr Just updated the PR. Please help check.
1. Merge the latest branch.
2. Handle case "Return to the profile page if there's no notes to preview upon deleting one."

**Comment by ArekChr** [comment]
@DylanDylann, I've encountered a new regression issue. When accessing Private Notes via a deep link, the app unexpectedly re-opens the Private Notes after entering and saving a note.

Steps to Reproduce:

1. Open any Private Note.
2. Copy the link for editing.
3. Delete the Private Note from input and save.
4. Log out of the app.
5. Paste the copied link.
6. Log back in.
7. Enter data in Private Notes and hit 'Save'.

After these steps, the input field for Private Notes appears again, which seems unintended.

This issue is not reproducible on staging.

https://github.com/Expensify/App/assets/24796318/93718d72-ab0f-468a-bbb7-1de619ae45c9

File: src/pages/ProfilePage.js [hunk 3]
```diff
@@ -255,7 +255,9 @@ function ProfilePage(props) {
                                 title={`${props.translate('privateNotes.title')}`}
                                 titleStyle={styles.flex1}
                                 icon={Expensicons.Pencil}
-                                onPress={() => Navigation.navigate(ROUTES.PRIVATE_NOTES_LIST.getRoute(props.report.reportID))}
+                                onPress={() => {
+                                    ReportUtils.navigateToPrivateNotes(props.report, props.session);
+                                }}
```
Comment (ArekChr) [inline src/pages/ProfilePage.js:260]: nit:
```suggestion
                                onPress={() => ReportUtils.navigateToPrivateNotes(props.report, props.session)}
```

File: src/pages/ReportDetailsPage.js [hunk 4]
```diff
@@ -138,13 +138,15 @@ function ReportDetailsPage(props) {
                 translationKey: 'privateNotes.title',
                 icon: Expensicons.Pencil,
                 isAnonymousAction: false,
-                action: () => Navigation.navigate(ROUTES.PRIVATE_NOTES_LIST.getRoute(props.report.reportID)),
+                action: () => {
+                    ReportUtils.navigateToPrivateNotes(props.report, props.session);
+                },
```
Comment (ArekChr) [inline src/pages/ReportDetailsPage.js:143]: nit: 
```suggestion
                action: () => ReportUtils.navigateToPrivateNotes(props.report, props.session),
```

**Comment by DylanDylann** [comment]
@ArekChr Updated PR to fix the issue you`ve mentioned. 
- RCA: After saving the private notes, it will navigate to the private notes list page but the first time the useEffect `s callback is called, the private notes list is still empty, which leads to navigating back to the edit page
- Solution: Use setimeout to delay 300ms before deciding whether the private notes list is empt or not
- Result:

[Screencast from 21-11-2023 11:54:32.webm](https://github.com/Expensify/App/assets/141406735/bd41b0bf-76b6-450e-b8f9-ac0cc28a1dda)

File: src/pages/ProfilePage.js (hunk 3 above)
Comment (DylanDylann) [reply src/pages/ProfilePage.js]: Updated

File: src/pages/ReportDetailsPage.js (hunk 4 above)
Comment (DylanDylann) [reply src/pages/ReportDetailsPage.js]: Updated

**Comment by DylanDylann** [comment]
@ArekChr Currently, the initial private note is something like {123456: {note: ''}}. So, I am not sure whether changing the initial value to undefined leads to regression or not

File: src/libs/ReportUtils.js [hunk 5]
```diff
@@ -4284,6 +4285,19 @@ function shouldDisableWelcomeMessage(report, policy) {
     return isMoneyRequestReport(report) || isArchivedRoom(report) || !isChatRoom(report) || isChatThread(report) || !PolicyUtils.isPolicyAdmin(policy);
 }
 
+/**
+ * @param {Object} report
```
Comment (ArekChr) [inline src/libs/ReportUtils.js:4289]:
```suggestion
/**
 * Navigates to the appropriate screen based on the presence of a private note for the current user.
 * @param {Object} report
```

File: src/pages/PrivateNotes/PrivateNotesListPage.js [hunk 6]
```diff
@@ -98,14 +100,30 @@ function PrivateNotesListPage({report, personalDetailsList, session}) {
...
+            }
+            Navigation.navigate(ROUTES.PRIVATE_NOTES_EDIT.getRoute(report.reportID, session.accountID));
+        }, CONST.ANIMATED_TRANSITION);
+        return () => {
+            if (!navigateToEditPageTimeoutRef.current) {
+                return;
+            }
+            clearTimeout(navigateToEditPageTimeoutRef.current);
+        };
+    }, [report.privateNotes, report.reportID, session.accountID, isFocused]);
```
Comment (ArekChr) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:126]:
```suggestion
    }, [report.privateNotes, report.reportID, session.accountID, isFocused]);
    
```

File: src/pages/PrivateNotes/PrivateNotesListPage.js [hunk 7]
```diff
@@ -98,14 +100,30 @@ function PrivateNotesListPage({report, personalDetailsList, session}) {
...
-                iconType: CONST.ICON_TYPE_AVATAR,
-                action: () => Navigation.navigate(ROUTES.PRIVATE_NOTES_VIEW.getRoute(report.reportID, accountID)),
+                action: () => Navigation.navigate(ROUTES.PRIVATE_NOTES_EDIT.getRoute(report.reportID, accountID)),
                 brickRoadIndicator: privateNoteBrickRoadIndicator(accountID),
+                note: lodashGet(privateNote, 'note', ''),
+                disabled: Number(session.accountID) !== Number(accountID),
             }))
             .value();
     }, [report, personalDetailsList, session, translate]);
+    const isFocused = useIsFocused();
```
Comment (ArekChr) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:111]:
```suggestion

    const isFocused = useIsFocused();
```

File: src/pages/PrivateNotes/PrivateNotesListPage.js [hunk 8]
```diff
@@ -98,14 +100,30 @@ function PrivateNotesListPage({report, personalDetailsList, session}) {
...
 
+    const navigateToEditPageTimeoutRef = React.useRef(null);
+    useEffect(() => {
```
Comment (ArekChr) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:113]: Let's move this `useEffect` between to line 61

**Comment by ArekChr** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/pages/PrivateNotes/PrivateNotesListPage.js [hunk 9]
```diff
@@ -98,14 +100,30 @@ function PrivateNotesListPage({report, personalDetailsList, session}) {
...
+                return;
+            }
+            Navigation.navigate(ROUTES.PRIVATE_NOTES_EDIT.getRoute(report.reportID, session.accountID));
+        }, CONST.ANIMATED_TRANSITION);
+        return () => {
+            if (!navigateToEditPageTimeoutRef.current) {
+                return;
+            }
+            clearTimeout(navigateToEditPageTimeoutRef.current);
+        };
```
Comment (ArekChr) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:125]:
```suggestion
        const navigateToEditPageTimeout = setTimeout(() => {
            if (_.some(report.privateNotes, (item) => item.note) || !isFocused) {
                return;
            }
            Navigation.navigate(ROUTES.PRIVATE_NOTES_EDIT.getRoute(report.reportID, session.accountID));
        }, CONST.ANIMATED_TRANSITION);
        
        return () => {
            clearTimeout(navigateToEditPageTimeout);
        };
```

File: src/pages/PrivateNotes/PrivateNotesListPage.js [hunk 10]
```diff
@@ -98,14 +100,30 @@ function PrivateNotesListPage({report, personalDetailsList, session}) {
...
+                action: () => Navigation.navigate(ROUTES.PRIVATE_NOTES_EDIT.getRoute(report.reportID, accountID)),
                 brickRoadIndicator: privateNoteBrickRoadIndicator(accountID),
+                note: lodashGet(privateNote, 'note', ''),
+                disabled: Number(session.accountID) !== Number(accountID),
             }))
             .value();
     }, [report, personalDetailsList, session, translate]);
+    const isFocused = useIsFocused();
 
+    const navigateToEditPageTimeoutRef = React.useRef(null);
```
Comment (ArekChr) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:112]: Ref is not needed, we can use local variable in useEffect
```suggestion
```

**Comment by ArekChr** [comment]
@DylanDylann please update with latest main

**Comment by DylanDylann** [comment]
@ArekChr Fixed all your comments and merge latest main, please help check

**Comment by ArekChr** [comment]
@DylanDylann, after removing private notes and opening other private notes that are not empty, the app redirects us to edit private notes, but should it redirect us to the private notes list?

https://github.com/Expensify/App/assets/24796318/43a39126-ee4a-420e-99a3-b2152b30f576

**Comment by DylanDylann** [comment]
@ArekChr It seems that I cannot reproduce the bug that you mentioned https://github.com/Expensify/App/pull/31344#issuecomment-1827506051. Here is the steps:
1. Make sure we have 2 report A and B with private notes
2. Open report A => report A `s detail page => private notes => remove private notes
3. Open report B => report B `s detail page => private notes => it will navigate user to report B `s private notes list (expected)

**Comment by ArekChr** [comment]
Retesting

**Review by ArekChr — APPROVED** [review]
After retesting all good, thanks for PR!

**Comment by techievivek** [comment]
Will test this today.

**Review by techievivek — APPROVED** [review]
Looks good and tests well with multiple notes.


https://github.com/Expensify/App/assets/35863227/b4cbb6e4-0a9a-4240-a2e0-e1e7d50a4d78

**Comment by chiragsalian** [comment]
Issue found [here](https://github.com/Expensify/App/issues/32302) while testing. Can you fix the code for this case @DylanDylann? More details [here](https://github.com/Expensify/App/issues/32302#issuecomment-1834719926).

**Comment by brunovjk** [comment]
We just resolved an issue that seems to be present since the [update of PrivateNotesListPage.js](https://github.com/DylanDylann/App/blob/de6787e296941c18a846fff0c077d940089bcf55/src/pages/PrivateNotes/PrivateNotesListPage.js#L135-L136) here, where it was added `<Text ... 'privateNotes.personalNoteMessage ...`. This `<Text ..` should be added inside the `<ScrollView` to _"roll along"_

**Pull request merged**